enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+	// "MatchRepoDigestOrExact" means that the identity in the signature must be in the same repository as the image identity if the image identity is referenced by a digest. Otherwise, the identity in the signature must be the same as the image identity.
+	// "MatchRepository" means that the identity in the signature must be in the same repository as the image identity.
+	// "ExactRepository" means that the identity in the signature must be in the same repository as a specific identity specified by "repository".
+	// "RemapIdentity" means that the signature must be in the same as the remapped image identity. Remapped image identity is obtained by replacing the "prefix" with the specified “signedPrefix” if the the image identity matches the specified remapPrefix.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum=MatchRepoDigestOrExact;MatchRepository;ExactRepository;RemapIdentity
+	// +kubebuilder:validation:Required
+	MatchPolicy IdentityMatchPolicy `json:"matchPolicy,omitempty"`
+	// exactRepository is required if matchPolicy is set to "ExactRepository".
+	// +optional
+	PolicyMatchExactRepository *ExactRepository `json:"exactRepository,omitempty"`
+	// remapIdentity is required if matchPolicy is set to "RemapIdentity".
+	// +optional
+	PolicyMatchRemapIdentity *RemapIdentity `json:"remapIdentity,omitempty"`
+}
+
+// IdentityMatchPolicy defines the type of matching for setting "matchPolicy".
+// Valid values are "MatchRepoDigestOrExact", "MatchRepository", "ExactRepository", "RemapIdentity". When omitted, the default value is "MatchRepoDigestOrExact".
+type IdentityMatchPolicy string
+
+type PolicyMatchExactRepository struct {
+	// repository is the reference of the image identity to be matched.
+	// +kubebuilder:validation:Required
+	Repository string `json:"repository,omitempty"`
+}
+
+type PolicyMatchRemapIdentity struct {
+	// prefix is the prefix of the image identity to be matched.
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix,omitempty"`
+	// signedPrefix is the prefix of the image identity to be matched in the signature.
+	// +kubebuilder:validation:Required
+	SignedPrefix string `json:"signedPrefix,omitempty"`
+}
+
+const (
+	IdentityMatchPolicyMatchRepoDigestOrExact IdentityMatchPolicy = "MatchRepoDigestOrExact"
+	IdentityMatchPolicyMatchRepository        IdentityMatchPolicy = "MatchRepository"
+	IdentityMatchPolicyExactRepository        IdentityMatchPolicy = "ExactRepository"
+	IdentityMatchPolicyRemapIdentity          IdentityMatchPolicy = "RemapIdentity"
+)
+
+// +k8s:deepcopy-gen=true
+type ImagePolicyStatus struct {
+
+	// policyJSON contains the whole policy applied to the kubernetes namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// conditions provide details on the status of the gatherer job.
```

Comment:
Is this comment correct?

There are a number of validation tags that normally accompany a conditions list, please make sure these are added (you'll see them on other conditions lists)